### PR TITLE
Add bounds check to Random getNext

### PR DESCRIPTION
**Pull request by ben-albrecht**
Add a halting bounds check to `RandomStreamInterface.getNext(min, max)` and `RandomStreamInterface.getNext(resultType, min, max)`.

### CODE REVIEW

**Comment by ben-albrecht** [comment]
After discussing with @ronawho, we decided it would be best to make this a halting error rather than a throwing error for performance reasons.

File: modules/standard/Random.chpl [hunk 1]
```diff
@@ -857,10 +857,13 @@ module Random {
            the interval [`min`, `max`] can be constructed in this way.
 
        */
-      proc getNext(min: eltType, max:eltType): eltType {
+      proc getNext(min: eltType, max:eltType): eltType throws {
```
Comment (cassella) [inline modules/standard/Random.chpl]: Does it still need to be `throws` now that the added error is a halt?

File: test/library/standard/Random/RandomStreamInterface/getNextTestHalts.good [hunk 2]
```diff
@@ -0,0 +1 @@
+getNextTestHalts.chpl:11: error: halt reached - Cannot generate random numbers within empty range: [2, 1]
```
Comment (ronawho) [inline test/library/standard/Random/RandomStreamInterface/getNextTestHalts.good]: Needs a skipif or suppressif for --fast (i.e. test/types/range/deitz/test_range_iterate_error2.suppressif)

File: modules/standard/Random.chpl (hunk 1 above)
Comment (ben-albrecht) [reply modules/standard/Random.chpl]: Does `chplspell` detect non-throwing `throws` functions now too? :)

Comment (cassella) [reply modules/standard/Random.chpl]: Just wait til you see what else is in the version 2 spell checker!

File: test/library/standard/Random/RandomStreamInterface/getNextTestHalts.skipif [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# --fast sets --no-checks
+#
+# --fast sets --no-checks and this test is explicitly checking that one
+# of our runtime checks is working correctly. There's been debate as to
+# whether tests like this should be suppressed or skipped. For now we
+# suppress with the argument that we're testing that --fast actually
```
Comment (mppf) [inline test/library/standard/Random/RandomStreamInterface/getNextTestHalts.skipif]: but you skip it?

File: modules/standard/Random.chpl (hunk 1 above)
Comment (ben-albrecht) [reply modules/standard/Random.chpl]: It turns out the introduction of `format()` introduced a throwing function to this function body, so I guess you'll have to fix that in `chplspell 2.1`.

File: modules/standard/Random.chpl [hunk 4]
```diff
@@ -843,7 +843,7 @@ module Random {
       }
       /*
         Return the next random value but within a particular range.
-        Returns a number in [`min`, `max`] (inclusive).
+        Returns a number in [`min`, `max`] (inclusive). Halts if ``min > max``.
```
Comment (mppf) [inline modules/standard/Random.chpl]: Maybe "Halts if checks are enabled and min > max".

**Pull request merged**